src/fabtrimino.c

```diff
@@ -0,0 +1,250 @@
+#include "fabtrimino.h"
+
+struct square
+{
+    enum shape shape;
+};
+
+struct fabtrimino
+{
+    enum shape shape;
+    struct square*** matrix;
+    struct vector* pos;
+};
+
+
+void nullify_squares_matrix(struct square*** mat, size_t size)
+// TODO: catch and print error when size > matrix_size
+// (maybe create struct for matrix with size)
+{
+    for (int i = 0; i < size; i++)
```

Comment:
I generally recommend using `size_t` defined in `stddef.h` or one of the `stdint.h` types over plain `int`. The reason is that int is signed which brings in a bunch of different undefined behavior and int's size can be a bit weird if you care about portability. In your case here, `size_t` is what I consider the superior type.

More details available here:  https://github.com/RAttab/slides/blob/master/how-to-c.org#ints